lib/zipkin-tracer/hostname_resolver.rb

```diff
@@ -0,0 +1,61 @@
+module ZipkinTracer
+  # Resolves hostnames in the endpoints of the annotations.
+  # Resolving hostnames is a very expensive operation. We want to store them raw in the main thread
+  # and resolve them in a different thread where we do not affect execution times.
+  class HostnameResolver
+    def spans_with_ips(spans)
+      host_to_ip = hosts_to_ipv4(spans)
+
+      each_annotation(spans) do |annotation|
+        hostname = annotation.host.ipv4
+        if hostname.to_s =~ /\A[a-z]/ #hostnames start with letters, else we already have an IP
```

Comment:
Missing space between `#` and the comment

hostname is always lower case right?